### ISSUE TITLE
Document window.fetch workarounds in cy.server + cy.route + Network Requests doc

### ISSUE THREAD

**Issue by brian-mann** [cypress-io/cypress-documentation#238]
People are very confused about this, why it doesn't show up in the command log and why its not being stubbed.

Need to call this out specifically in those docs.

**Comment by felquis** [comment]
Past week it was logging and matching API pathnames perfectly without any workaround.

Then I changed the react-scripts dev server to serve the static build with the npm serve CLI as in `serve --single --no-clipboard --listen 3001 ./build`

With this change cypress couldn't even load pages properly, besides it loads the page on the right, in the left panel it says it didn't load. It was solved by using `chromeWebSecurity: false` https://github.com/cypress-io/cypress/issues/4142#issuecomment-578243243

But now I can't match routes which was working fine before, I load my static on localhost:3001 and the API is an external staging server https://api-staging.domain.com

I added the hack to reset win.fetch 

```js
Cypress.on("window:before:load", (win) => {
    win.fetch = null;
});
```
Nothing changed. I guess I'm missing some fundamental piece of this whole story.
